sinks/gcm/driver.go

```diff
@@ -303,7 +319,22 @@ func (self *gcmDriver) PushMetrics(metrics []Metric) error {
 		return err
 	}
 
-	return sendRequest(fmt.Sprintf("https://www.googleapis.com/cloudmonitoring/v2beta2/projects/%s/timeseries:write", self.project), self.token, request)
+	const requestAttempts = 3
+	for i := 0; i < requestAttempts; i++ {
+		err = sendRequest(fmt.Sprintf("https://www.googleapis.com/cloudmonitoring/v2beta2/projects/%s/timeseries:write", self.project), self.token, request)
+		if err != nil {
+			glog.Warningf("[GCM] Push attempt %d failed: %v", i, err)
+		} else {
+			break
+		}
+	}
+	if err != nil {
+		prettyRequest, _ := json.MarshalIndent(request, "", "  ")
+		glog.Warningf("[GCM] Pushing %d metrics \n%s\n failed: %v", len(request.Timeseries), string(prettyRequest), err)
```

Comment:
I'm planning to instrument this with Prometheus metrics (we can for all of Heapster). To know how many timeseries we've pushed and the error rate. We can keep the last N of these in Validate too.